zookeeper-server/src/main/java/org/apache/zookeeper/client/ConnectStringParser.java

```diff
@@ -73,7 +54,11 @@ public ConnectStringParser(String connectString) {
                 }
                 host = host.substring(0, pidx);
             }
-            serverAddresses.add(InetSocketAddress.createUnresolved(host, port));
+            try {
+                serverAddresses.add(new InetSocketAddress(InetAddress.getByName(host), port));
+            } catch (UnknownHostException e) {
```

Comment:
we shouldn't catch this

Reply:
createUnresolved() doesn't throw because it defers lookup, and UnknownHostException is checked, so wrapping a RuntimeException instead. As it was originally, the client probably would've thrown an unrelated error about not having any servers to talk to.